### PR TITLE
[IMP/ADD][12.0] Fleet Management: Improved modules flow  and removed unnecessary code.

### PR DESCRIPTION
**Pull request by ChavdaRiddhi**

### CODE REVIEW

File: fleet_operations/models/fleet_extended.py [hunk 1]
```diff
@@ -136,14 +122,17 @@ def name_get(self):
 
     @api.model
     def name_search(self, name='', args=[], operator='ilike', limit=100):
-        # Overwritten this method for the bypass base domain.
+        """Overwritten this method for the bypass base domain."""
         vehicle_ids = self.search(args, limit=limit)
         return vehicle_ids.name_get()
```
Comment (YogendrasinhChavda) [inline fleet_operations/models/fleet_extended.py:127]: @ChavdaRiddhi what is the use case of this name_search method here?

File: fleet_operations/models/fleet_extended.py [hunk 2]
```diff
@@ -504,18 +514,16 @@ class ColorHistory(models.Model):
 
     @api.multi
     def copy(self, default=None):
+        """Method copy."""
         if not default:
             default = {}
         raise Warning(_('You can\'t duplicate record!'))
         return super(ColorHistory, self).copy(default=default)
```
Comment (YogendrasinhChavda) [inline fleet_operations/models/fleet_extended.py:521]: @ChavdaRiddhi Test and improve copy method.
I mean no need raise exception we can restrict using the Access right.

File: fleet_operations/models/fleet_service.py [hunk 3]
```diff
@@ -62,19 +51,102 @@ def get_vehicle_info(self):
...
+        for service in self:
+            deposit_inv_ids = self.env['account.invoice'].search([
+              ('vehicle_service_id', '=', service.id), ('type', '=', 'out_invoice'),
+                ('state', 'in', ['draft', 'open', 'in_payment'])
+                ])
+            if deposit_inv_ids:
+                raise Warning(_("Deposit invoice is already Pending\n"
+                    "Please proceed that deposit invoice first"))
+
+        self.ensure_one()
```
Comment (YogendrasinhChavda) [inline fleet_operations/models/fleet_service.py:75]: It should be the beginning of the method.

File: fleet_operations/models/fleet_service.py [hunk 4]
```diff
@@ -62,19 +51,102 @@ def get_vehicle_info(self):
...
+        for service in self:
+            deposit_inv_ids = self.env['account.invoice'].search([
+              ('vehicle_service_id', '=', service.id), ('type', '=', 'out_refund'),
+                ('state', 'in', ['draft', 'open', 'in_payment'])
+                ])
+            if deposit_inv_ids:
+                raise Warning(_("Deposit Return invoice is already Pending\n"
+                    "Please proceed that deposit invoice first"))
+
+        self.ensure_one()
```
Comment (YogendrasinhChavda) [inline fleet_operations/models/fleet_service.py:117]: self.ensure_one() should be the beginning of the method.

**Comment by YogendrasinhChavda** [comment]
@ChavdaRiddhi I merged this pull request, but still code optimization and improvements are needed.
plz, do it.

**Comment by ChavdaRiddhi** [comment]
@YogendrasinhChavda ok..i will check and fix it.

**Pull request merged**